### PR TITLE
Recurring reminders on the thank you page

### PR DESCRIPTION
**Pull request by tomrf1**
Currently all users see this, even if they have signed up.
If a user is signed in then we could find out from MDAPI if they already have a reminder, but I'm leaving this to a follow up task.

A new option for monthly:
Screen Shot 2021-03-10 at 11 32 26

The recurring request:
Screen Shot 2021-03-10 at 11 32 44

The single request still works:
Screen Shot 2021-03-08 at 11 10 42

### CODE REVIEW

File: support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx [hunk 1]
```diff
@@ -51,15 +51,22 @@ const privacyTextLink = css`
...
-  signup: ReminderSignup;
+interface OneOffReminderChoice {
+  type: 'ONE_OFF';
+  signup: {
+    reminderPeriod: string;
+    reminderOption: string;
+  };
   label: string;
 }
+type ReminderChoice = RecurringReminderChoice | OneOffReminderChoice;
```
Comment (tomrf1) [inline support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:69]: I initially tried to modify the existing `ReminderChoice` model with `signup: RecurringReminderSignup | OneOffReminderSignup`.
But flow gets confused, even if they have a `type` field

File: support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx [hunk 2]
```diff
@@ -80,17 +87,35 @@ const getDefaultLabel = (date: Date, monthsUntilDate: number, now: Date) => {
   const year =
     now.getFullYear() === date.getFullYear() ? '' : ` ${date.getFullYear()}`;
 
-  return `in ${monthsUntilDate} months (${month}${year})`;
+  return `in ${monthsUntilDate} months only (${month}${year})`;
+};
+
+const getRecurringReminderChoice = (): ReminderChoice => {
+  const now = new Date();
+  const startMonth = now.getMonth() + (now.getDate() >= 20 ? 1 : 0); // TODO - is this right?
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:95]: I think it'll be `2 : 1`

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:95]: https://github.com/guardian/support-dotcom-components/blob/main/src/lib/reminderFields.ts#L8

Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:95]: that's the equivalent in SDC

Comment (tomrf1) [reply support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouSupportReminder.jsx:95]: oops, you're quite right!

**Review by TomPretty — APPROVED** [review]
Looks great to me 👍

**Pull request merged**